client/src/App.js

```diff
@@ -312,6 +334,18 @@ export default class NormalDisplay extends React.Component {
             running: true,
             logs: undefined
         });
+        //validate if journal ready
+        if(!this.state.journalReady){
+            const journalReady = await this.checkJournalReady(this.state.journalUrl);
```

Comment:
From what I see above, `checkJournalReady` will throw if not ready. You will never have `journalReady: false`. Is this intended that way? To throw on not ready?

Reply:
yes, I am still debating if I should have the `journalReady: false` check below, given `checkJournalReady()` only returns true and the exception is handled in its catch block. its harmless, but possibly redundant.